pkg/recipes/terraform/config/providers/types.go

```diff
@@ -34,13 +34,37 @@ type Provider interface {
 	BuildConfig(ctx context.Context, envConfig *recipes.Configuration) (map[string]any, error)
 }
 
-// GetSupportedTerraformProviders returns a map of Terraform provider names to provider config builder.
-// Providers represent Terraform providers for which Radius generates custom provider configurations.
-// For example, the Azure subscription id is added to Azure provider config using Radius Environment's Azure provider scope.
-func GetSupportedTerraformProviders(ucpConn sdk.Connection, secretProvider *ucp_provider.SecretProvider) map[string]Provider {
+// GetUCPConfiguredTerraformProviders returns a map of Terraform provider names with configuration details stored in UCP, to provider config builder.
+// These providers represent Terraform providers for which Radius generates custom provider configurations based on credentials stored with UCP
+// and providers configured on the Radius environment. For example, the Azure subscription id is added to Azure provider config using Radius Environment's Azure provider scope.
+func GetUCPConfiguredTerraformProviders(ucpConn sdk.Connection, secretProvider *ucp_provider.SecretProvider) map[string]Provider {
 	return map[string]Provider{
 		AWSProviderName:        NewAWSProvider(ucpConn, secretProvider),
 		AzureProviderName:      NewAzureProvider(ucpConn, secretProvider),
 		KubernetesProviderName: &kubernetesProvider{},
 	}
 }
+
+// GetRecipeProviderConfigs returns the Terraform provider configurations for Terraform providers
+// specified under the RecipeConfig/Terraform/Providers section under environment configuration.
+func GetRecipeProviderConfigs(ctx context.Context, envConfig *recipes.Configuration) map[string]any {
+	providerConfigs := make(map[string]any)
+
+	// If the provider is not configured, or has empty configuration, skip this iteration
+	if envConfig != nil && envConfig.RecipeConfig.Terraform.Providers != nil {
+		for provider, config := range envConfig.RecipeConfig.Terraform.Providers {
+			if len(config) > 0 {
+				configList := make([]any, 0)
+
+				// Retrieve configuration details from 'AdditionalProperties' property and add to the list.
+				for _, configDetails := range config {
+					configList = append(configList, configDetails.AdditionalProperties)
+				}
```

Comment:
Should check if `AdditionalProperties` is non-empty/non-nil?

```suggestion
				for _, configDetails := range config {
    if configDetails.AdditionalProperties != nil && len(configDetails.AdditionalProperties) > 0 {
					configList = append(configList, configDetails.AdditionalProperties)
					}
				}
```

Reply:
@kachawla this is allowed here to accomodate 'default provider configuration' where no details are input. Tests are added for this case both in types_test.go and here: https://github.com/radius-project/radius/pull/7189/files#diff-c5c89506f1ba2fe9bd8ecf4eedc856e93714c251c295c32a15d875f5924df3bdR515-R530
a example resulting main.tf.json is here: [pkg/recipes/terraform/config/testdata/providers-envrecipedefaultconfig.tf.json ](https://github.com/radius-project/radius/pull/7189/files#diff-f607148044721443b79901397b6794542895307f824510496a86ad52cd5e1b61R12-R18)
I did a manual test to see provider configuration load successfully.

Reply:
Sounds good, thank you. During manual testing, in addition to checking that the config was loaded successfully, could you also confirm that terraform apply runs successfully for this scenario?

Reply:
Also, can you please expand on "default provider configuration"? I'm not sure what we mean by this. Feel free to point me to the right section in the design if it is captured in the doc already.

Reply:
yes, confirming, terraform apply runs successfully for this scenario.

Reply:
Why do we want to populate empty config though?

Reply:
There is no concept of default empty configuration afaik, let's add this nil check unless there is a terraform documentation that mentions otherwise.

Reply:
added nil check.

Reply:
From the doc - "Terraform assumes an empty default configuration for any provider that is not explicitly configured." - this is already handled by Terraform. By adding empty configuration here, we are overriding any non-empty config that a user might have in their module. Unless there is a strong use case for passing empty config, I wouldn't recommend doing this.